mint-auditor/src/convert/block_audit_data.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Convert to/from mc_mint_auditor_api::BlockAuditData.
+
+use crate::BlockAuditData;
+use std::{
+    collections::{BTreeMap, HashMap},
+    iter::FromIterator,
+};
+
+/// Convert BlockAuditData --> mc_mint_auditor_api::BlockAuditData
+impl From<&BlockAuditData> for mc_mint_auditor_api::BlockAuditData {
+    fn from(src: &BlockAuditData) -> Self {
+        let mut dst = mc_mint_auditor_api::BlockAuditData::new();
+        dst.set_balance_map(HashMap::from_iter(src.balance_map.clone().into_iter()));
```

Comment:
IIUC this can use `collect()`

```suggestion
        dst.set_balance_map(src.balance_map.iter().cloned().collect());
```
Might not need `.cloned()`

Reply:
It needed it otherwise it's `(&u32, &u64)` and the compiler complains. I'll see about collect.